scan/tls_handshake.go

```diff
@@ -342,3 +346,29 @@ func certSigAlgsScanByCipher(addr, hostname string) (grade Grade, output Output,
 	}
 	return
 }
+
+// ecCurveScan returns the elliptic curves supported by the host.
+func ecCurveScan(addr, hostname string) (grade Grade, output Output, err error) {
+	allCurves := allCurvesIDs()
+	curves := make([]tls.CurveID, len(allCurves))
+	copy(curves, allCurves)
+	var supportedCurves []string
+	for len(curves) > 0 {
+		var curveIndex int
+		_, curveIndex, _, err = sayHello(addr, hostname, nil, curves, tls.VersionTLS12, nil)
+		if err != nil {
+			// This case is expected, because eventually we ask only for curves the server doesn't support
+			if err == errHelloFailed {
+				err = nil
+				break
+			}
+			return
+		}
+		curveID := curves[curveIndex]
+		supportedCurves = append(supportedCurves, tls.Curves[curveID])
+		curves = append(curves[:curveIndex], curves[curveIndex+1:]...)
+	}
+	output = supportedCurves
+	grade = Good
```

Comment:
It would be nice if we had a list of "Bad" curves [here](https://github.com/cloudflare/cf-tls/blob/master/tls/cfsslscan_common.go#L476)